src/utils/futures.rs

```diff
@@ -96,6 +99,29 @@ pub struct RemoteThread {
     arbiter: Option<actix::Addr<actix::Arbiter>>,
 }
 
+/// Spawning the future on the remote failed.
+///
+/// This is an error caused by the remote, not the result or output of the future itself.
+/// In the future this could include loadshedding errors and the like.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum SpawnError {
+    /// The future was cancelled, e.g. because the remote was dropped.
+    Canceled,
+    /// The future exceeded its timeout.
+    Timeout,
+}
+
+/// Dropping terminates the RemoteThread, cancelling all futures.
+impl Drop for RemoteThread {
+    fn drop(&mut self) {
+        // Without sending the StopArbiter message the arbiter thread keeps running even if
+        // you drop the arbiter.
+        if let Some(ref addr) = self.arbiter {
+            addr.do_send(actix::msgs::StopArbiter(0));
+        }
+    }
+}
+
```

Comment:
This is wrong, because the `RemoteThread` can be cloned which only clones the address of the arbiter dropping a clone would terminate the arbiter while there was still another one.  Need to wrap the arbiter into an inner `Arc` to make this work.